### PR TITLE
Add analytics for layout overwrite/revert; provide event category to Sentry breadcrumbs

### PR DESCRIPTION
**Pull request by jtbandes**
**User-Facing Changes**
None

**Description**
Add analytics for new actions added in #1679. Add `type` and `category` to Sentry breadcrumbs instead of sending the event name as the category.

Related: #1620

### CODE REVIEW

File: packages/studio-base/src/services/IAnalytics.ts [hunk 1]
```diff
@@ -23,13 +30,94 @@ enum AppEvent {
...
+}
+
+export function getEventBreadcrumbType(event: AppEvent): SentryBreadcrumbType {
+  switch (event) {
+    case AppEvent.APP_INIT:
+      return SentryBreadcrumbType.DEFAULT;
+
+    case AppEvent.PLAYER_CONSTRUCTED:
+    case AppEvent.PLAYER_INITIALIZED:
+      return SentryBreadcrumbType.TRANSACTION;
```
Comment (jhurliman) [inline packages/studio-base/src/services/IAnalytics.ts:94]: `NAVIGATION`?

File: packages/studio-base/src/services/IAnalytics.ts [hunk 2]
```diff
@@ -23,13 +30,94 @@ enum AppEvent {
...
+
+    case AppEvent.PLAYER_PLAY:
+    case AppEvent.PLAYER_SEEK:
+    case AppEvent.PLAYER_SET_SPEED:
+    case AppEvent.PLAYER_PAUSE:
+      return SentryBreadcrumbType.USER;
+
+    case AppEvent.PLAYER_CLOSE:
+      return SentryBreadcrumbType.TRANSACTION;
```
Comment (jhurliman) [inline packages/studio-base/src/services/IAnalytics.ts:103]: `NAVIGATION`?

**Review by jhurliman — APPROVED** [review]

File: packages/studio-base/src/services/IAnalytics.ts (hunk 1 above)
Comment (jtbandes) [reply packages/studio-base/src/services/IAnalytics.ts:94]: Not very clear to me the intended difference between `transaction` (a "tracing event") and `navigation` (a "UI transition") ... which one is a change in player? The UI doesn't really transition...the user has just made a selection.

Comment (jtbandes) [reply packages/studio-base/src/services/IAnalytics.ts:94]: Btw I'm not sure the `type` really does anything in the sentry UI other than change an icon.

**Pull request merged**